### ISSUE TITLE
About inference...

### ISSUE THREAD

**Issue by zha-hengfeng** [clovaai/mxfont#8]
I noticed that there is a source_path in inference.ipynb that needs to specify a ttf your data folder does not contain, does it use the same font as your lffont's content_font. If not, how do I get it?

**Comment by 8uos** [comment]
Hi.
We used LF-Font's content font as the source font to generate the result images reported in our paper. 
You can download it from [here](https://chinesefontdesign.com/font-housekeeper-song-ming-typeface-chinese-font-simplified-chinese-fonts.html).
Sorry for the inconvenience-we could not include the font file because of the license issue.
Also, it is okay to use any font as the source font because our model is trained to recognize the character from various fonts. (We recommend using an "ordinary" font like ming.)

**Comment by zha-hengfeng** [comment]
Thank you for your answer, it is a very great work. But our inference on the collected handwriting found that mxfont did not work as well as lffont, what could be the reason for this result.
![image](https://user-images.githubusercontent.com/43509883/127761120-c5b61af2-f516-493d-b47c-a5ce0bd00c44.png)

**Comment by SanghyukChun** [comment]
@zha-hengfeng 
Hi, a few-shot font generation task is a very challenging task, and we are making some progress towards better stylization. We tried directly utilizing component information (or radial information) in LF-Font, while it has drawbacks such as (1) unable to generate unseen components during training time (because we have to use "direct condition" for each component (2) we need component labels in the inference time, this is not a big problem, but definitely it harms the practical usability.

In MX-Font, we fix two problems by implicitly utilizing component labels by algorithmic improvements (details are in our paper). However, it does not mean that MX-Font is always better than LF-Font in terms of the visual quality of generated glyphs. Well, although I believe that MX-Font is better than LF-Font in terms of the visual quality as well, we don't strongly argue that MX-Font always shows much better quality than LF-Font.

Instead, we showed that MX-Font can generate "cross-lingual" fonts while LF-Font cannot (e.g., generating Korean fonts by the model trained with Chinese fonts). Also, measuring the visual quality of the generated fonts are not trivial; we applied four types of different metrics (accuracies, user study, LPIPS, FID) You can see that LF-Font shows slightly better FID / LPIPS performances than MX-Font in CHN -> CHN (while MX-Font still performs the best in Acc, User preferences), but MX-Font is the only method working on the cross-lingual scenario (CHN -> KOR)

![result_table](https://user-images.githubusercontent.com/4919315/128588493-6c908bf5-b0ca-4952-8782-f5b7bef5bfc8.png)

To sum up, we don't argue that MX-Font is always better than LF-Font. It could be depending on many situations, and in the paper, we focused on the cross-lingual scenario. The performance could depend on the input complexity, characters, ...
If you need really good font generation results, we suggest you apply many-shot methods (fine-tuning on the reference characters) such as [zi2zi](https://github.com/kaonashi-tyc/zi2zi)

**Comment by SanghyukChun** [comment]
Closing the issue, assuming the answer resolves the problem.
Please re-open the issue as necessary.
